components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java

```diff
@@ -232,4 +245,203 @@ public static String prependOperationScenarioToErrorCode(String exceptionErrorCo
+            if (StringUtils.isBlank(body)) {
+                String errorCode =
+                        I18nEmailUtil.prependOperationScenarioToErrorCode(
+                                I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE.getCode(),
+                                I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+                throw new NotificationTemplateManagerClientException(errorCode,
+                        I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE.getMessage());
+            }
+            if (StringUtils.isNotBlank(subject) || StringUtils.isNotBlank(footer)) {
+                String errorCode =
+                        I18nEmailUtil.prependOperationScenarioToErrorCode(
+                                I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE_CONTENT.getCode(),
+                                I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+                throw new NotificationTemplateManagerClientException(errorCode,
+                        I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_SMS_TEMPLATE_CONTENT.getMessage());
+            }
+        } else {
+            if (StringUtils.isBlank(subject) || StringUtils.isBlank(body)) {
+                String errorCode =
+                        I18nEmailUtil.prependOperationScenarioToErrorCode(
+                                I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_EMAIL_TEMPLATE.getCode(),
+                                I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+                throw new NotificationTemplateManagerClientException(errorCode,
+                        I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_EMAIL_TEMPLATE.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Validate the display name of the notification template.
+     *
+     * @param displayName Display name
+     * @throws NotificationTemplateManagerClientException Invalid notification template name
+     */
+    public static void validateDisplayNameOfTemplateType(String displayName)
+            throws NotificationTemplateManagerClientException {
+
+        if (StringUtils.isBlank(displayName)) {
+            String errorCode =
+                    I18nEmailUtil.prependOperationScenarioToErrorCode(
+                            I18nMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_TEMPLATE_NAME.getCode(),
+                            I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+            throw new NotificationTemplateManagerClientException(errorCode,
+                    I18nMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_TEMPLATE_NAME.getMessage());
+        }
+        /*Template name can contain only alphanumeric characters and spaces, it can't contain registry invalid
+        characters*/
+        String[] whiteListPatterns = {TEMPLATE_REGEX_KEY};
+        String[] blackListPatterns = {REGISTRY_INVALID_CHARS};
```

Comment:
Shall we change the variable names here?

Reply:
Fixed with: https://github.com/wso2-extensions/identity-event-handler-notification/pull/231/commits/bb2c28dfeef67b93afd36b89564e4d3338b8bf45